_posts/2020-08-18-deepfake/deepfake.Rmd

```diff
@@ -0,0 +1,257 @@
+
+Our approach consists of the following consequent steps:
+
+- read all the videos
+- capture and extract images from the videos
+- detect faces from the exracted images
+- crop the faces
+- build an image classification model with Keras
+
+OpenCV is a computer vision library that includes:
+
+- Facial recognition technology
+- [Motion tracking](https://en.wikipedia.org/wiki/Video_tracking)
+- [Augmented reality](https://en.wikipedia.org/wiki/Augmented_reality)
+- etc.
+
+On the other hand, `magick` is the open-source image-processing library that will help to read and extract useful features from video datasets:
+
+- Read video files
+- Extracts images per second of video
+- Crop the faces from the images
+
+Before we go into a detailed explanation, readers should know that there is no need to copy-paste code chunks. Because at the end of the post one can find a link to Google Colab with GPU acceleration. This kernel allows everyone to run and reproduce the same results.
+
+
+## Data exploration
+
+[The dataset](https://www.kaggle.com/c/deepfake-detection-challenge/overview/description) that we are going to analyze is provided by AWS, Facebook, Microsoft, the Partnership on AI’s Media Integrity Steering Committee, and various academics.
+
+It contains both real and AI-generated fake videos. The total size is over 470 GB. However, the sample 4 GB dataset is separately available.
+
+## Frame extraction
+
+The videos in the folders are in the format of _mp4_ and have various lengths. Our task is to determine the number of images to capture per second of a video. We usually took 1-3 fps for every video.
+
+> Note: Set fps to NULL if you want to extract all frames.
+
+```{r eval=F, echo=T}
+video = magick::image_read_video("aagfhgtpmv.mp4",fps = 2)
+vid_1 = video[[1]]
+vid_1 = magick::image_read(vid_1) %>% image_resize('1000x1000')
+```
+
+<center>
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = "[Deepfake detection challenge](https://www.kaggle.com/c/deepfake-detection-challenge/data)"}
+knitr::include_graphics("files/frame_1.jpg")
+```
+
+</center>
+
+We saw just the first frame. What about the rest of them?
+
+```{r, eval=TRUE, echo=FALSE, layout="l-body", fig.cap = "[Deepfake detection challenge](https://www.kaggle.com/c/deepfake-detection-challenge/data)"}
+knitr::include_graphics("files/self.gif")
+```
+
+
+Looking at the gif one can observe that some fakes are very easy to differentiate, but a small fraction looks pretty realistic. This is another challenge during data preparation.
+
+## Face detection
+
+We can read all the files or take a sample. And afterward, detect and extract faces. 
+
+There is a very well-known library OpenCV^[[Opencv](https://opencv.org/)] which is designed for computer vision tasks. R has also some bindings to OpenCV, so extracting faces will not be a challenge for us.
```

Comment:
I think with the introduction of OpenCV you've added, this paragraph can be deleted now